src/lib.rs

```diff
@@ -133,38 +150,112 @@ where
-        ctrl4 |= range.bits() << 4;
+            ctrl4 &= !FS_MASK;
+            // Write in new full-scale to highest 4 bits
+            ctrl4 |= range.bits() << 4;
 
             ctrl4
         })
     }
 
-    /// Read the current range
+    /// Read the current full-scale
     pub fn get_range(&mut self) -> Result<Range, Error<E>> {
         let ctrl4 = self.read_register(Register::CTRL4)?;
         let fs = (ctrl4 >> 4) & 0x03;
 
-        match Range::try_from(fs) {
-            Ok(range) => Ok(range),
-            Err(_) => Err(Error::InvalidRange),
+        Range::try_from(fs).map_err(|_| Error::InvalidRange)
+    }
+
+    /// Set `REFERENCE` register
+    pub fn set_ref(&mut self, reference: u8) -> Result<(), Error<E>> {
+        self.write_register(Register::REFERENCE, reference)
+    }
+
+    /// Read the `REFERENCE` register
+    pub fn get_ref(&mut self) -> Result<u8, Error<E>> {
+        self.read_register(Register::REFERENCE)
+    }
+
+    /// Data status.
+    /// `STATUS_REG`: as
+    /// DataStatus { zyxor: `ZYXOR`, xyzor: (`XOR`, `YOR`, `ZOR`),
+    ///              zyxda: `ZYXDA`, xyzda: (`XDA`, `YDA`, `ZDA`) }
+    pub fn get_status(&mut self) -> Result<DataStatus, Error<E>> {
+        let stat = self.read_register(Register::STATUS)?;
+
+        Ok(DataStatus {
+            zyxor: (stat & ZYXOR) != 0,
+            xyzor: ((stat & XOR) != 0, (stat & YOR) != 0, (stat & ZOR) != 0),
+            zyxda: (stat & ZYXDA) != 0,
+            xyzda: ((stat & XDA) != 0, (stat & YDA) != 0, (stat & ZDA) != 0),
+        })
+    }
+
+    /// Temperature sensor enable.
+    /// `TEMP_CGF_REG`: `TEMP_EN`, the BDU bit in `CTRL_REG4` is also set
+    pub fn enable_temp(&mut self, enable: bool) -> Result<(), Error<E>> {
+        self.register_xset_bits(Register::TEMP_CFG, TEMP_EN, enable)?;
+
+        // enable block data update (required for temp reading)
+        if enable {
+            self.register_xset_bits(Register::CTRL4, BDU, true)?;
         }
+
+        Ok(())
+    }
+
+    /// Temperature sensor data.
+    /// `OUT_TEMP_H`, `OUT_TEMP_L`
+    pub fn get_temp_out(&mut self) -> Result<(i8, u8), Error<E>> {
+        let out_l = self.read_register(Register::OUT_ADC2_L)?;
+        let out_h = self.read_register(Register::OUT_ADC2_H)?;
+
+        Ok((out_h as i8, out_l))
+    }
+
+    /// Temperature sensor data converted to f32.
+    /// `OUT_TEMP_H`, `OUT_TEMP_L`
+    pub fn get_temp_outf(&mut self) -> Result<f32, Error<E>> {
+        let (out_h, out_l) = self.get_temp_out()?;
+        // 10-bit resolution
+        let value = ((out_h as i16) << 2) | ((out_l as i16) >> 6);
+
+        Ok((value as f32) * 0.25)
+    }
+
+    /// Reboot memory content.
+    /// `CTRL_REG5`: `BOOT`
+    pub fn reboot(&mut self, reboot: bool) -> Result<(), Error<E>> {
+        self.register_xset_bits(Register::CTRL5, BOOT, reboot)
+    }
+
+    /// In boot,
+    /// `CTRL_REG5`: `BOOT`
+    pub fn in_boot(&mut self) -> Result<bool, Error<E>> {
```

Comment:
is_in_boot is more idiomatic for booleans

Reply:
Good eye! I'll get that changed, thanks.